### ISSUE TITLE
BUG: Co-simulation failing to stop at simEnd 

### ISSUE THREAD

**Issue by sibyjackgrove** [tdcosim/TDcoSim#10]
**Describe the bug**
The simulation is failing to if simEnd is 20 s and the exception **'failed to run dynamic simulation'** is raised from line 66 in default_dynamic_procedure.py.

**To Reproduce**
Run the dynamic co-simulation with simEnd: 20 s, faultOn at 1.5s, faultOff at 1.5s  at 1.667 s

**Expected behavior**
Co-simulation should stop at simEnd 20s.

**Screenshots**
Screenshot of load from buses with DSS containing DER penetration from result.out file.

![image](https://user-images.githubusercontent.com/25213730/67692919-df532e00-f96e-11e9-80bf-17c902c9f119.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - PSS/E Version: 33.3.0

**Additional context**
Was doing dynamic cosimulation with 5 nodes at 50 % penetration.

**Comment by yim0331** [comment]
@sibyjackgrove Could you update the config.json file? The problem is not generated in my machine with the reproduction process you mentioned above.

**Comment by yim0331** [comment]
Relative Issue: https://github.com/tdcosim/TDcoSim/issues/1

**Comment by sibyjackgrove** [comment]
@yim0331 please find config file:

```
"cosimHome": "..\\tdcosim",
    "psseConfig":{
        "rawFilePath":"..\\SampleData\\TNetworks\\118bus\\case118.raw",
        "dyrFilePath":"..\\SampleData\\TNetworks\\118bus\\case118.dyr"        
    },
    "openDSSConfig":{        
        "defaultFeederConfig":{
            "filePath":["..\\SampleData\\DNetworks\\123Bus\\case123ZIP.dss"],
            "solarFlag":0,
            "solarPenetration":0.0
        },
        "manualFeederConfig":{
            "nodes": [
                {
                    "nodenumber": 11,
                    "filePath": ["..\\SampleData\\DNetworks\\123Bus\\case123ZIP.dss"],
                    "solarFlag":1,
                    "solarPenetration":0.2,
                    "DERParameters":{
                        "pvderScale": 1,
                        "V_LV0": 0.50,   
                        "V_LV1": 0.70,
                        "V_LV2": 0.88,
                        "t_LV0_limit": 1.0,  
                        "t_LV1_limit": 10.0,  
                        "t_LV2_limit": 20.0,
                        "V_HV1":1.06,
                        "V_HV2":1.12,
                        "t_HV1_limit": 3.0,
                        "t_HV2_limit": 0.016,
                        "VRT_INSTANTANEOUS_TRIP": false,
                        "VRT_MOMENTARY_CESSATION": true,
                        "OUTPUT_RESTORE_DELAY": 0.5,
                        "power_rating":50,
                        "SteadyState":true
                    }
                },
                {
                    "nodenumber": 14,
                    "filePath": ["..\\SampleData\\DNetworks\\123Bus\\case123ZIP.dss"],
                    "solarFlag":1,
                    "solarPenetration":0.2,
                    "DERParameters":{
                        "pvderScale": 1,
                        "V_LV0": 0.50,   
                        "V_LV1": 0.70,
                        "V_LV2": 0.88,
                        "t_LV0_limit": 1.0,  
                        "t_LV1_limit": 10.0,  
                        "t_LV2_limit": 20.0,
                        "V_HV1":1.06,
                        "V_HV2":1.12,
                        "t_HV1_limit": 3.0,
                        "t_HV2_limit": 0.016,
                        "VRT_INSTANTANEOUS_TRIP": false,
                        "VRT_MOMENTARY_CESSATION": true,
                        "OUTPUT_RESTORE_DELAY": 0.5,
                        "power_rating":50,
                        "SteadyState":true
                    }
                },
                {
                    "nodenumber": 15,
                    "filePath": ["..\\SampleData\\DNetworks\\123Bus\\case123ZIP.dss"],
                    "solarFlag":1,
                    "solarPenetration":0.2,
                    "DERParameters":{
                        "pvderScale": 1,
                        "V_LV0": 0.50,   
                        "V_LV1": 0.70,
                        "V_LV2": 0.88,
                        "t_LV0_limit": 1.0,  
                        "t_LV1_limit": 10.0,  
                        "t_LV2_limit": 20.0,
                        "V_HV1":1.06,
                        "V_HV2":1.12,
                        "t_HV1_limit": 3.0,
                        "t_HV2_limit": 0.016,
                        "VRT_INSTANTANEOUS_TRIP": false,
                        "VRT_MOMENTARY_CESSATION": true,
                        "OUTPUT_RESTORE_DELAY": 0.5,
                        "power_rating":50,
                        "SteadyState":true
                    }
                }
            ]
        }
    },
    "simulationConfig":{
        "simType":"dynamic",
        "dynamicConfig":{
            "events":{
                "1":{
                     "time":1.5,
                    "type":"faultOn",
                    "faultBus":15,
                    "faultImpedance":[0.0,-10000]
                },
                "2":{
                    "type":"faultOff",
                    "time":1.667,
                    "faultBus":15
                },
                "3":{
                    "type":"simEnd",
                    "time":25.0
                }
            }
        },
        "staticConfig":{
            "loadShape": [0.81518164, 0.77456979, 0.74237094, 0.72439771, 0.71946463,
            0.72891013, 0.75212237, 0.76833652, 0.783174  , 0.79468451,
            0.80634799, 0.80370937, 0.8016826 , 0.79288719, 0.79437859,
             0.79720841, 0.82271511, 0.92470363, 1.0, 0.99021033,
             0.97101338, 0.94042065, 0.88397706, 0.81097514]
        },
        "protocol":"loose_coupling"
    },
    "outputConfig":{
        "outputfilename": "output.csv",
        "type": "csv"
    }

```

**Comment by yim0331** [comment]
Hi Siby,

The test is ended at the expected time (25.00823) I send my config.json and result.xels.
Please let me know if I missed anything to regenerate this issue.

**Comment by sibyjackgrove** [comment]
Ok, this means this bug only occurs when there is simulation failure similar to #13.

**Comment by yim0331** [comment]
This issue is fixed by Commit: bd4d9e5468e0322e3dba269f4f04a1e695d7b1ba
Although PSSE failed run, the simulation will be end on time.
